ci-operator/step-registry/cucushift/upgrade/toimage/cucushift-upgrade-toimage-commands.sh

```diff
@@ -0,0 +1,201 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+# Extract oc binary which is supposed to be identical with target release
+# parameters: 1-release pull spec
+function extract_oc(){
+    set -x
+    echo "Extracting oc: $1"
+    if ! oc adm release extract --command='oc' "${1}" && chmod +x ./oc; then
+        echo >&2 "Failed to extract oc binary, exiting" && exit 1
+    fi
+    set +x
+}
+
+# Get the target upgrades release, by default, RELEASE_IMAGE_LATEST is the target release
+# If it's serial upgrades then override-upgrade file will store the release and overrides RELEASE_IMAGE_LATEST
+# override-upgrade file expects a comma separated releases list like target_release1,target_release2,...
+function get_target_release() {
+    echo "RELEASE_IMAGE_INITIAL is ${RELEASE_IMAGE_INITIAL}"
+    echo "RELEASE_IMAGE_LATEST is ${RELEASE_IMAGE_LATEST}"
+    TARGET_RELEASES=("${RELEASE_IMAGE_LATEST}") &&
+    if [[ -f "${SHARED_DIR}/override-upgrade" ]]; then
+        local release_string; release_string="$(< "${SHARED_DIR}/override-upgrade")"
+        # shellcheck disable=SC2207
+        TARGET_RELEASES=($(echo "$release_string" | tr ',' ' ')) 
+        echo "Overriding upgrade target to ${TARGET_RELEASES[*]}"
+    fi
+}
+
+# Check if a release image signed
+# Parameter: 1- by-digest pull spec
+function validate_signed_image() {
+    digest="$(echo "$1" | cut -f2 -d@)"
+    algorithm="$(echo "${digest}" | cut -f1 -d:)"
+    hash="$(echo "${digest}" | cut -f2 -d:)"
+    local response; response=$(curl --silent --output /dev/null --write-out "%{http_code}" "https://mirror.openshift.com/pub/openshift-v4/signatures/openshift/release/${algorithm}=${hash}/signature-1")
+    if [[ "$response" == "200" ]]; then
+        echo "True"
+    else
+        echo "False"
+    fi
+}
+
+# Check if admin ack is required before upgrade
+# Parameters: 1-source release minor version, 2-target release minor version
+function admin_ack() {
+    set -x
+    if [[ $1 -eq $2 ]]; then
+        echo "Upgrade between z-stream version does not require admin ack"
+    else
+        local out; out="$(./oc -n openshift-config-managed get configmap admin-gates -o json | jq -r ".data")"
```

Comment:
Suggestion: you can move this entire block out of the `if-else` block, so it's easier to read. eg:
```
if [[ $1 -eq $2 ]]; then
    echo "Upgrade between z-stream version does not require admin ack"
    return
fi

local out; out="$(./oc -n openshift-config-managed get configmap admin-gates -o json | jq -r ".data")"
...
```